package-config.cmake

```diff
@@ -174,3 +176,41 @@ install(
     DESTINATION "${CONFIG_INSTALL_DIR}"
 )
 ENDMACRO(SETUP_PROJECT_PACKAGE_FINALIZE)
+
+#.rst:
+# .. command:: PROJECT_INSTALL_COMPONENT(COMPONENT [EXTRA_MACRO cmake_code])
+#
+#   Generates CMake componentConfig.cmake and Targets files so users can call::
+#
+#     find_package(MyPackage COMPONENT component)
+#
+#   :param EXTRA_MACRO: optional argument. `cmake_code` will be appended to
+#          the generated *Config.cmake*.
+#
+macro(PROJECT_INSTALL_COMPONENT COMPONENT)
+  cmake_parse_arguments(PARSED_ARGN "" "EXTRA_MACRO" "" ${ARGN})
+
+  set(namespace "${PROJECT_NAME}::")
```

Comment:
I think for components it might be interesting to have the option to specify a different export namespace